ntoskrnl/mm/ARM3/virtual.c

```diff
@@ -733,8 +738,8 @@ MiDeleteVirtualAddresses(IN ULONG_PTR Va,
 
         if (Va > EndingAddress) return;
 
-        /* Otherwise, we exited because we hit a new PDE boundary, so start over */
-        AddressGap = FALSE;
+        /* Check if we exited the loop regularly */
+        AddressGap = (PointerPte != MiAddressToPte(Va));
```

Comment:
Question: If this is just for handling the `Use this to detect address gaps` part in the modified if-block above, wouldn't it then be simpler to do a `AddressGap = TRUE;` inside that block?
Or, is this condition also covering the fact that `PointerPte` is also increased in the main do-while loop, and an address gap can develop there as well?

Reply:
First there is not neccessarily a gap, so you can't just set it to TRUE, but you need to compare the incremented PointerPte with the new address. Also when you set it there, it will break out of the loop and then reset it back to false.